### PR TITLE
Improving SS as a source of truth (#56)

### PR DESCRIPTION
**Pull request by quinnwai**
Addressing #56

### Thinking through user story
1. I have changes I want to make to local, so I...
```bash
cp Secrets-local Secrets-local_test
make push-secrets DEPLOY=local_test
```

2. Once it's updated in SS, other people can test those changes by...
```bash
make fetch-secrets DEPLOY=local_test` (writes to `Secrets-local_test/` dir)
rm -r Secrets/
ln -s Secrets-local_test Secrets

# TODO: some level of testing on their local, eg
make clean && make update && make local 
```

3. With other's approval, I update the true local with...
```bash
make push-secrets DEPLOY=local_test
rm -r Secrets-local_test/
```

## Descriptions / commit messages
Tried to create a description to add to each action, but the actions in the Audit (ie history tab) are not editable. Let me know if you think commit messages / descriptions are still important or if we should just leave it

### CODE REVIEW

**Comment by matthewpeterkort** [comment]
Need to test but overall this looks good to me.

Tested and working. LGTM

File: .gitignore [hunk 1]
```diff
@@ -17,4 +17,4 @@ user.yaml
 # ohsu
 venv/
 .SSToken.*
-Secrets-local/
\ No newline at end of file
+Secrets-*/
```
Comment (lbeckman314) [inline .gitignore:20]: Nice catch

File: Makefile [hunk 2]
```diff
@@ -137,4 +119,31 @@ help:	## Show this help message
...
 
-.PHONY: debug deploy clean check-clean zip help change-context
\ No newline at end of file
+.PHONY: debug deploy clean check-clean zip help change-context
+
+
+#################
+# SECRET SERVER #
+#################
+# venv will be created if it doesn't exist
```
Comment (lbeckman314) [inline Makefile:128]: Great additions with [create-venv](https://github.com/ACED-IDP/gen3-helm/blob/e995f7d4b5df6d982e4a66a5ba36cbf097e42a35/Makefile#L56-L64) and [check-venv](https://github.com/ACED-IDP/gen3-helm/blob/e995f7d4b5df6d982e4a66a5ba36cbf097e42a35/Makefile#L66-L71)!

File: SSClient.py [hunk 3]
```diff
@@ -66,7 +69,7 @@ def conv_shorthand(env: str) -> str:
     return env
 
 
-def match_env_with_id(env: str, id: int):
+def match_env_with_id(env: str, id: Optional[int]):
     """Secret Server expects an 'id' in order to get data from a secret.
        To do this you need to"""
```
Comment (lbeckman314) [inline SSClient.py]: Nice job adding documentation to the methods! This particular text might be flying over my head — does "To do this you need to" refer to passing the `id` to Secret Server?

**Review by lbeckman314 — DISMISSED** [review]

File: Makefile (hunk 2 above)
Comment (quinnwai) [reply Makefile:128]: thanks! those were already there courtesy of Matthew but I just added more comments

File: SSClient.py (hunk 3 above)
Comment (matthewpeterkort) [reply SSClient.py]: Secret server stores secrets as records with a unique id. In order to access a specific secret you need to know the id. That's why the list secrets command exists. 

Second sentence is a fragment and can be deleted

Comment (quinnwai) [reply SSClient.py]: Updated

**Review by matthewpeterkort — APPROVED** [review]

**Pull request merged**